### PR TITLE
Whoami shouldnt redirect

### PR DESCRIPTION
**Pull request by hatched**
Adds the ability to stop 401 redirects in the bakery. Fixes: #1445

### CODE REVIEW

**Comment by hatched** [comment]
#### To QA

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4615/
Test PASSed.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/test-jujulib-environment.js [hunk 1]
```diff
@@ -184,8 +184,11 @@ describe('jujulib environment manager', function() {
   it('identifies the current user', function(done) {
     var currentUser = {user: 'test'};
     var bakery = {
-      sendGetRequest: function(path, data, success, failure) {
+      sendGetRequest: function(path, success, failure, redirect) {
+        console.log(arguments);
```
Comment (huwshimi) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/test-jujulib-environment.js]: Was this intended to be left in?

Comment (hatched) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/test-jujulib-environment.js]: Nope, oops

**Comment by huwshimi** [comment]
QA OK :+1:

**Comment by hatched** [comment]
Thanks! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4621/
Test PASSed.

**Pull request merged**